kicad: 5.1.9 -> 5.1.10; plus bunch of cleanup

evils opened the pull request:
###### Motivation for this change
upstream my local changes

supersedes / closes #120129

###### Things done
cleanup version string post patch removal
use latest wxPython package for kicad-unstable
link through the new /share/metainfo
bump kicad-unstable to latest commit
bump kicad to latest release

bit of cleanup
including adding a bunch of inputs in response to configuration warnings (all linux specific)

expose the new address and threads sanitation options

revert https://github.com/NixOS/nixpkgs/pull/100066/files#diff-42931fe739a41421fbba1e00b70d4ec50f620a254b86430a40df5e2a355cd820R54
as embedding the date causes a large rebuild after midnight (UTC)
additionally, reverted such a change in the libraries, as they change less often than the date does and just generates store garbage

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
  - `/nix/store/2rgrws9iz17rbzd3ni7pky4ps5d3zsal-kicad-5.1.9	 6751202712`
  - `/nix/store/s5nzi6n6w88yxvarha6yc3v7iwsrzs7z-kicad-5.1.10	 6751000128`
  - `/nix/store/zdzrngcbl71wbg7wsvc6zbrravc4vr6c-kicad-unstable-2020-12-23	 6914682792`
  - `/nix/store/0pl7p5176k287zxqz8m6gdhny4kmspwh-kicad-unstable-8513ca974c	 6949715408`
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

evils committed:
kicad-unstable: set KICAD6 env vars

evils committed:
kicad-unstable: use wxPython 4.1

evils committed:
kicad-unstable: 2020-12-23 -> 2021-05-13

evils committed:
kicad: add sanitizeAddress & sanitizeThreads options

only kicad 5.99 or later are affected
was briefly -DKICAD_SANITIZE=ON
are mutually exclusive according to a compiler error

evils committed:
kicad: avoid rebuild on date change

the date changes more often than most of the inputs
the date can also change on the base without other changes
  this causes a substantial rebuild for no reason

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/default.nix:
```diff
@@ -193,14 +194,31 @@ stdenv.mkDerivation rec {
   ]
-  ++ optionals (with3d) [ "--set-default KISYS3DMOD ${packages3d}/share/kicad/modules/packages3d" ]
+  ++ optionals (stable && with3d) [ "--set-default KISYS3DMOD ${packages3d}/share/kicad/modules/packages3d" ]
+  ++ optionals (!stable)
+  [
```
```suggestion
  ++ optionals (!stable) [
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/default.nix:
```diff
@@ -193,14 +194,31 @@ stdenv.mkDerivation rec {
+    "--set-default KICAD6_TEMPLATE_DIR ${templates}/share/kicad/template"
+    "--prefix KICAD6_TEMPLATE_DIR : ${symbols}/share/kicad/template"
+    "--prefix KICAD6_TEMPLATE_DIR : ${footprints}/share/kicad/template"
+  ]
+  ++ optionals (!stable && with3d)
+  [
```
```suggestion
  ] ++ optionals (!stable && with3d) [
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/default.nix:
```diff
@@ -193,14 +194,31 @@ stdenv.mkDerivation rec {
     "--set-default KICAD_TEMPLATE_DIR ${templates}/share/kicad/template"
     "--prefix KICAD_TEMPLATE_DIR : ${symbols}/share/kicad/template"
     "--prefix KICAD_TEMPLATE_DIR : ${footprints}/share/kicad/template"
   ]
```
```suggestion
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/default.nix:
```diff
@@ -193,14 +194,31 @@ stdenv.mkDerivation rec {
     "--prefix KICAD_TEMPLATE_DIR : ${footprints}/share/kicad/template"
   ]
-  ++ optionals (with3d) [ "--set-default KISYS3DMOD ${packages3d}/share/kicad/modules/packages3d" ]
+  ++ optionals (stable && with3d) [ "--set-default KISYS3DMOD ${packages3d}/share/kicad/modules/packages3d" ]
```
```suggestion
   ] ++ optionals (stable && with3d) [ "--set-default KISYS3DMOD ${packages3d}/share/kicad/modules/packages3d" ]
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/default.nix:
```diff
@@ -193,14 +194,31 @@ stdenv.mkDerivation rec {
-
+    # required to open a bug report link in firefox-wayland
+    "--set-default MOZ_DBUS_REMOTE 1"
+  ]
+  ++ optionals (stable)
+  [
```
```suggestion
  ] ++ optionals stable [
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/default.nix:
```diff
@@ -146,28 +148,27 @@ let
 
   python = python3;
-  wxPython = python.pkgs.wxPython_4_0;
+  wxPython = if (stable)
```
```suggestion
  wxPython = if stable
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/default.nix:
```diff
@@ -146,28 +148,27 @@ let
 
   inherit pname;
-  version = kicadVersion;
+  version = if (stable) then kicadVersion else builtins.substring 0 10 src.src.rev;
```
```suggestion
  version = if stable then kicadVersion else builtins.substring 0 10 src.src.rev;
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/base.nix:
```diff
@@ -90,16 +109,42 @@ stdenv.mkDerivation rec {
       "-DKICAD_STDLIB_DEBUG=ON"
       "-DKICAD_USE_VALGRIND=ON"
     ]
+    ++ optionals (sanitize) [
+      "-DKICAD_SANITIZE=ON"
+    ]
+    ++ optionals (withWayland) [
```
```suggestion
    ] ++ optionals sanitize [
      "-DKICAD_SANITIZE=ON"
    ] ++ optionals withWayland [
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/base.nix:
```diff
@@ -15,58 +15,77 @@
-      --replace "unknown" "${builtins.substring 0 10 src.rev}-1" \
-      --replace "${version}" "${version}-1"
+      --replace "unknown" "${builtins.substring 0 10 src.rev}"
+  ''
+  + optionalString (withWayland) ''
```
```suggestion
  '' + optionalString (withWayland) ''
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/base.nix:
```diff
@@ -90,16 +109,42 @@ stdenv.mkDerivation rec {
+    pkg-config
+    lndir
+  ]
+  # wanted by configuration on linux, doesn't seem to affect performance
```
But closure size and build time?

evils replied:
i'll check that out

evils replied:
no effect on closure size, i'll note that in the comment (i seem to recall the hash of the binary (sha256sum) (binary not wrapper) changed)

evils replied:
and still ~15m build time

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/base.nix:
```diff
@@ -15,58 +15,77 @@
 stdenv.mkDerivation rec {
   pname = "kicad-base";
-  version = kicadVersion;
+  version = if (stable) then kicadVersion else builtins.substring 0 10 src.rev;
```
```suggestion
  version = if stable then kicadVersion else builtins.substring 0 10 src.rev;
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/base.nix:
```diff
@@ -15,58 +15,77 @@
 }:
 
 assert lib.asserts.assertMsg (!(withOCE && stdenv.isAarch64)) "OCE fails a test on Aarch64";
 assert lib.asserts.assertMsg (!(withOCC && withOCE))
   "Only one of OCC and OCE may be enabled";
```
```suggestion
assert lib.asserts.assertMsg (!(withOCC && withOCE)) "Only one of OCC and OCE may be enabled";
```

evils replied:
hey, that's not even my change!

evils commented:
@SuperSandro2000 thanks for the many suggested changes
i can't help think it'd have been less trouble on your part to just add a comment suggesting style points

i also think i would have received such a suggestion better...

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/base.nix:
```diff
@@ -15,58 +15,77 @@
+, libdatrie
+, libxkbcommon
+, epoxy
+, dbus_daemon
```
```suggestion
, dbus
```

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/base.nix:
```diff
@@ -90,16 +109,43 @@ stdenv.mkDerivation rec {
+    libdatrie
+    libxkbcommon
+    epoxy
+    dbus_daemon
```
```suggestion
    dbus.daemon
```

evils replied:
thanks

evils commented:
bumped unstable from 2021-04-20 to 2021-04-22

SuperSandro2000 commented on pkgs/applications/science/electronics/kicad/base.nix:
```diff
@@ -15,58 +15,77 @@
+      --replace "unknown" "${builtins.substring 0 10 src.rev}"
+  ''
+  + optionalString (withWayland) ''
+    substituteInPlace kicad/kicad.cpp \
+      --replace "x11" "wayland"
+    substituteInPlace common/single_top.cpp \
+      --replace "x11" "wayland"
+    substituteInPlace  libs/kiplatform/gtk/environment.cpp \
+      --replace "x11" "wayland"
```
This could be simplified.

```suggestion
    for f in kicad/kicad.cpp common/single_top.cpp libs/kiplatform/gtk/environment.cpp; do
      substituteInPlace $f --replace "x11" "wayland"
    done
```

SuperSandro2000 reviewed (commented):
can you cleanup the history a bit?

evils commented:
> can you cleanup the history a bit?

i could maybe squash `kicad: rearrange / sort inputs` in somewhere, but it's not really part of another change, hence its own commit, same with the rest of them...

evils commented on pkgs/applications/science/electronics/kicad/base.nix:
```diff
@@ -15,58 +15,76 @@
+    done
   '';
 
   makeFlags = optional (debug) [ "CFLAGS+=-Og" "CFLAGS+=-ggdb" ];
```
i just spotted this should be `optionals`

i'll squash a few of the minor commits along with this correction into a "minor cleanup" commit

evils committed:
kicad: bunch of minor cleanup

squash of the following

kicad: makeFlags: optional -> optionals

kicad: remove -1 in version string, no more patch

should have been removed in a52e974cff8

kicad: remove berce as maintainer

kicad: gtk3 -> wxGTK.gtk

use wxGTK.gtk rather than gtk3

add graphviz as doxygen wants it (no idea if doxygen is required...)
  minor compilation warning

kicad: rearrange / sort inputs

separate optional deps and options

kicad: clear some configuration warnings

linux specific stuff that's wanted by something in config

kicad: fix opening bug report in firefox wayland

may be removed when kicad goes native?

evils commented:
the sanitize option was changed to two config options, address and thread sanitation
converted to draft while i sort that out

evils committed:
kicad-unstable: link through new /share/metainfo

evils commented:
changed sanitize option to sanitizeAddress and sanitizeThreads
added an assert because there's a compiler error noting they're mutually exclusive

linked through /share/metainfo, which is new, not bothering with an `if (!stable)` because it just results in an empty directory on the stable package

additionally, apparently kicad 5.1.10 was tagged 2 days ago
the i18n library is still missing that tag
otherwise i'd include it here
this shouldn't block merging

evils committed:
kicad: 5.1.9 -> 5.1.10

evils commented:
kicad 5.1.10 released, included that here
also bumped kicad-unstable to latest again
added assert to ensure withWayland is only attempted on kicad-unstable, as only that has the build option

evils commented:
removed the withWayland commit from this PR as it's experimental and currently failing

evils commented:
added some more cleanup, on cmakeFlags formatting
moved the sanitation options to base.nix and added one for the stable build (the sanitize build options are useless there)

evils commented:
bumped kicad-unstable again

@SuperSandro2000 sorry to ping you
but apparently i can't re-request a review due to your previous review being open?
despite both notes being marked as resolved and being outdated...

SuperSandro2000 commented:
> @SuperSandro2000 sorry to ping you
> but apparently i can't re-request a review due to your previous review being open?
> despite both notes being marked as resolved and being outdated...

Yeah,  the github review feature has some room for improvement. I'll try to take a look at this in the next days.

q3k reviewed (approved):
Drive-by, armchair, uneducated LGTM :).

evils commented:
bumped kicad-unstable again

evils commented:
bumped again to surpass #121741

evils commented:
bumped kicad-unstable again because they added a change that should improve compatibility with newer ngspice packages

evils commented:
@SuperSandro2000 you're probably busy with ZHF, but with how involved you are with this PR i don't think anyone else is going to bring it to merger

evils commented:
another kicad-unstable bump
and rebase to nixos-unstable (hence force push of all commits)

artemist commented:
The python console and 3d libraries are working fine for me on an x86_64 NixOS machine. I believe the code looks great except for that dbus_daemon to dbus.daemon suggestion (the dbus_daemon alias isn't used much in the codebase)

evils commented:
> except for that dbus_daemon to dbus.daemon suggestion (the dbus_daemon alias isn't used much in the codebase)

i'm not entirely clear what you mean there, i took the suggestion and dbus.daemon is used now, which seems to be what you're encouraging

artemist commented:
I might have misunderstood what GitHub was telling me. Oops!


On Fri, May 14, 2021, at 22:35, evils wrote:
> 

>> except for that dbus_daemon to dbus.daemon suggestion (the dbus_daemon alias isn't used much in the codebase)

> i'm not entirely clear what you mean there, i took the suggestion and dbus.daemon is used now, which seems to be what you're encouraging


> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/NixOS/nixpkgs/pull/119986#issuecomment-841586892>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAJLPDRGMBF3KI6Z2LRZZI3TNXMVNANCNFSM43JD4T7Q>.
